clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3489,9 +3506,26 @@ public void consumerBatchUnsubscribe(Set<PubSubTopicPartition> topicPartitionSet
 
   public abstract void consumerUnSubscribeAllTopics(PartitionConsumptionState partitionConsumptionState);
 
-  public void consumerSubscribe(PubSubTopicPartition pubSubTopicPartition, long startOffset, String kafkaURL) {
-    // TODO: pass special format of kafka url as input here when subscribe to the separate incremental push topic
+  /**
+   * This method will try to resolve actual topic-partition from input Kafka URL and subscribe to the resolved
+   * topic-partition.
+   */
+  public void consumerSubscribe(
+      PubSubTopic pubSubTopic,
+      PartitionConsumptionState partitionConsumptionState,
+      long startOffset,
+      String kafkaURL) {
+    PubSubTopicPartition resolvedTopicPartition =
+        resolveTopicPartitionWithKafkaURL(pubSubTopic, partitionConsumptionState, kafkaURL);
+    consumerSubscribe(resolvedTopicPartition, startOffset, kafkaURL);
+  }
+
+  void consumerSubscribe(PubSubTopicPartition pubSubTopicPartition, long startOffset, String kafkaURL) {
     String resolvedKafkaURL = kafkaClusterUrlResolver != null ? kafkaClusterUrlResolver.apply(kafkaURL) : kafkaURL;
+    if (!Objects.equals(resolvedKafkaURL, kafkaURL) && !isSeparatedRealtimeTopicEnabled()
+        && pubSubTopicPartition.getPubSubTopic().isRealTime()) {
+      return;
+    }
     final boolean consumeRemotely = !Objects.equals(resolvedKafkaURL, localKafkaServer);
     // TODO: Move remote KafkaConsumerService creating operations into the aggKafkaConsumerService.
     aggKafkaConsumerService
```

Comment:
This is a good catch. I think we should not expose this to AggKafkaConsumerService, so I made the logic tweak inside StorageUtilizationManager. It seems like this is the only place where these pause/resume is used. 
For reset, it seems like it will always on VT??? (This is what I got from code reading, I did not pay much attention to this API before)

Reply:
I was thinking if we could limit the scope of resolve topics for hybrid leader inside PartitionConsumptionState to let it figure out if we need to return 1 or 2 topics, then SIT consumer dealing functions (e.g consumerSubscribe) only handle the topics PCS returned.  The reason is there are a lot of interaction between SIT and AggkafkaConsumerService. AggKafkaConsumerService could only deal with one topic partition a time, that means SIT function or its related class should take a lot of consideration about ingestion from 1 or 2 topics.